modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuilder.java

```diff
@@ -133,71 +176,77 @@ private BuildIndexTask(
 
         @Override
         public void run() {
-            if (!busyLock.enterBusy()) {
-                return;
-            }
+            completedFuture(null)
+                    .thenCompose(unused -> inBusyLock(() -> {
+                        // At the time of creating the index, we should have already waited for the table to be created and its raft of
+                        // clients (services) to start for all partitions, so there should be no errors.
```

Comment:
probably something like
```suggestion
                        // At the time of index creation, table and raft services of all partitions 
                        // should be already started, so there should be no errors.
```

Reply:
Fix it